Sources/DatadogCrashReporting/PLCrashReporterIntegration/PLCrashReportFormatter.swift

```diff
@@ -0,0 +1,88 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import CrashReporter
+import Datadog
+
+/// A convenience interface of the`PLCrashReport` to enable its testability.
+internal protocol PLCrashReportType {
+    var timestamp: Date? { get }
+    var signalName: String? { get }
+    var signalCode: String? { get }
+    var exceptionName: String? { get }
+    var exceptionReason: String? { get }
+    var formattedStackTrace: String? { get }
+    var contextData: Data? { get }
+}
+
+extension PLCrashReport: PLCrashReportType {
+    var timestamp: Date? { systemInfo?.timestamp }
+    var signalName: String? { signalInfo?.name }
+    var signalCode: String? { signalInfo?.code }
+    var exceptionName: String? { hasExceptionInfo ? exceptionInfo?.exceptionName : nil }
+    var exceptionReason: String? { hasExceptionInfo ? exceptionInfo?.exceptionReason : nil }
+    var formattedStackTrace: String? { PLCrashReportTextFormatter.stringValue(for: self, with: PLCrashReportTextFormatiOS) }
+    var contextData: Data? { customData }
+}
+
+/// Formats the `PLCrashReport` to Datadog format (`DDCrashReport`).
+internal struct PLCrashReportFormatter {
+    private let unknown = "<unknown>"
+    private let unavailable = "<unavailable>"
+
+    func ddCrashReport(from crashReport: PLCrashReportType) -> DDCrashReport {
+        return DDCrashReport(
+            date: crashReport.timestamp,
+            type: readType(from: crashReport),
+            message: readMessage(from: crashReport),
+            stackTrace: readStackTrace(from: crashReport),
+            context: crashReport.contextData
+        )
+    }
+
+    // MARK: - Private
+
+    private func readType(from crashReport: PLCrashReportType) -> String {
+        return "\(crashReport.signalName ?? unknown) (\(crashReport.signalCode ?? unknown))"
+    }
+
+    private func readMessage(from crashReport: PLCrashReportType) -> String {
+        if crashReport.exceptionName != nil || crashReport.exceptionReason != nil {
+            // If the crash was caused by an uncaught exception
+            let exceptionName = crashReport.exceptionName // e.g. `NSInvalidArgumentException`
+            let exceptionReason = crashReport.exceptionReason // e.g. `-[NSObject objectForKey:]: unrecognized selector sent to instance 0x...`
+            return "Terminating app due to uncaught exception '\(exceptionName ?? unknown)', reason: '\(exceptionReason ?? unknown)'."
+        } else {
+            // Use signal description available in OS
+            guard let signalName = crashReport.signalName else { // e.g. SIGILL
```

Comment:
I'm wondering if for Swift runtime exceptions, we could leverage the (little) information available here:

https://developer.apple.com/documentation/xcode/diagnosing_issues_using_crash_reports_and_device_logs/identifying_the_cause_of_common_crashes/addressing_crashes_from_swift_runtime_errors

SIG TRAP 5.

Reply:
I added one more variant: _"Force try! Swift Error"_, which does:
```swift
try! throwException()
```
Got this on my iPhone X:

<img width="329" alt="Screenshot 2021-02-16 at 17 03 16" src="https://user-images.githubusercontent.com/2358722/108088547-e24cc300-7078-11eb-9e96-665193680009.png">

```
Hardware Model: iPhone10,6
Process: Example [24108]
...
Code Type: ARM-64
...
OS Version: iPhone OS 14.3 (18C66)
...
Exception Type: SIGTRAP
Exception Codes: #0 at 0x1a0eceaa8
```
It's somehow similar to what Apple doc says, but doesn't give `Termination Reason: Namespace SIGNAL, Code 0x5` 💭 ...